model-optimizer/mo/main.py

```diff
@@ -98,35 +98,44 @@ def print_argv(argv: argparse.Namespace, is_caffe: bool, is_tf: bool, is_mxnet:
 
 def get_moc_frontends(argv: argparse.Namespace):
     fem = argv.feManager
-    available_moc_front_ends = []
-    moc_front_end = None
-
-    # TODO: in future, check of 'use_legacy_frontend' in argv can be added here (issue 61973)
-    force_use_legacy_frontend = False
-
-    if fem and not force_use_legacy_frontend:
-        available_moc_front_ends = fem.get_available_front_ends()
-        if argv.input_model:
-            if not argv.framework:
-                moc_front_end = fem.load_by_model(argv.input_model)
-                # skip onnx frontend as not fully supported yet (63050)
-                if moc_front_end and moc_front_end.get_name() == "onnx":
-                    moc_front_end = None
-                if moc_front_end:
-                    argv.framework = moc_front_end.get_name()
-            elif argv.framework in available_moc_front_ends:
-                moc_front_end = fem.load_by_framework(argv.framework)
+
+    # Read user flags:
+    use_legacy_frontend = argv.use_legacy_frontend
+    use_new_frontend = argv.use_new_frontend
+
+    if not fem or use_legacy_frontend:
+        return None, []
+
+    available_moc_front_ends = fem.get_available_front_ends()
+
+    if not argv.framework and argv.input_model:
+        moc_front_end = fem.load_by_model(argv.input_model)
+        if not moc_front_end:
+            return None, available_moc_front_ends
+        argv.framework = moc_front_end.get_name()
+    elif argv.framework in available_moc_front_ends:
+        moc_front_end = fem.load_by_framework(argv.framework)
+    else:
+        return None, []
+
+    # Set which frontend to use by default, values should be 'new' or 'legacy'
+    frontend_defaults = {
+        'onnx': 'legacy',
+    }
+    # Disable MOC frontend if default is set to legacy and no user override
+    if frontend_defaults.get(moc_front_end.get_name()) == 'legacy' and not use_new_frontend:
+        moc_front_end = None
 
     return moc_front_end, available_moc_front_ends
 
 
 def arguments_post_parsing(argv: argparse.Namespace):
     moc_front_end, available_moc_front_ends = get_moc_frontends(argv)
 
-    is_tf, is_caffe, is_mxnet, is_kaldi, is_onnx =\
-        deduce_framework_by_namespace(argv) if not moc_front_end else [False, False, False, False, False]
+    is_tf, is_caffe, is_mxnet, is_kaldi, is_onnx, is_paddle =\
+        deduce_framework_by_namespace(argv) if not moc_front_end else [False, False, False, False, False, False]
 
-    if not any([is_tf, is_caffe, is_mxnet, is_kaldi, is_onnx]):
+    if not any([is_tf, is_caffe, is_mxnet, is_kaldi, is_onnx, is_paddle]):
         frameworks = ['tf', 'caffe', 'mxnet', 'kaldi', 'onnx']
         frameworks = list(set(frameworks + available_moc_front_ends))
         if argv.framework not in frameworks:
```

Comment:
can we remove 'is_paddle' (and related code to check .pdmodel extension) and insert here code like
```
if argv.use_legacy_frontend:
    if argv.framework is not None:
        raise Error('It is not possible to use the --use_legacy_frontend flag when processing a {} 
            model'.format(argv.framework)
    else: # framework is None
        raise Error('No suitable legacy frontend is found for model {}'.format(argv.input_model)  # Optionally add list of available_moc_front_ends
```
The goal is to reduce framework-dependent code in MO as much as possible for new frontends. MO shall not rely on knowledge that there is 'PaddlePaddle', 'Keras' or any other moc frontends

Reply:
Added changes as discussed offline.